diff --git a/src/schema.rs b/src/schema.rs
index 27db591..a42512f 100644
--- a/src/schema.rs
+++ b/src/schema.rs
@@ -39,12 +39,30 @@ table! {
         created_by -> Agent_id,
         created_at -> Timestamptz,
         deleted_at -> Nullable<Timestamptz>,
     }
 }
 
+table! {
+    use diesel::sql_types::*;
+    use crate::db::sql::*;
+
+    event_state_backward (id) {
+        id -> Uuid,
+    }
+}
+
+table! {
+    use diesel::sql_types::*;
+    use crate::db::sql::*;
+
+    event_state_forward (id) {
+        id -> Uuid,
+    }
+}
+
 table! {
     use diesel::sql_types::*;
     use crate::db::sql::*;
 
     room (id) {
         id -> Uuid,
@@ -56,13 +74,17 @@ table! {
     }
 }
 
 joinable!(adjustment -> room (room_id));
 joinable!(agent -> room (room_id));
 joinable!(event -> room (room_id));
+joinable!(event -> event_state_backward (id));
+joinable!(event -> event_state_forward (id));
 
 allow_tables_to_appear_in_same_query!(
     adjustment,
     agent,
     event,
+    event_state_backward,
+    event_state_forward,
     room,
 );
